File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        tot_sum=sum(skill)
        target = tot_sum//(len(skill)//2)
        i,j = 0, len(skill) - 1
        ans = 0
        while i < j:
            if skill[i] + skill[j] == target:
                ans = ans + skill[i] * skill[j] 
            else:
                return -1
            i += 1
            j -= 1

        return ans;        

File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(),skill.end());
        long long int sum = accumulate(skill.begin(),skill.end(),0);
        long long int ans = 0;

        long long int target = sum/(skill.size()/2);
        for(int i = 0, j = skill.size()-1; i<j; i++, j--) {
            if ((long long int )skill[i] + (long long int )skill[j] == target) {
                ans = ans + (skill[i]*skill[j]);
            } else {
                return -1;
            }
        }
        return ans;
    }
};